docs/browser.md

```diff
@@ -0,0 +1,400 @@
+
+#### Key Differences:
+- **Implementation file**: Runs in infinite loop for continuous load testing, accepts dynamic user credentials
+- **Spec file**: Runs once with hardcoded credentials for manual testing and validation
+
+### Running Manual Tests
+
+#### Prerequisites
+```bash
+cd browser
+make install
+```
+
+#### Running Individual Test Scenarios
+
+1. **Run all E2E tests**:
+   ```bash
+   npm run e2etest:run
+   ```
+
+2. **Run tests with UI (interactive mode)**:
+   ```bash
+   npm run e2etest:ui
+   ```
+
+3. **Debug tests step by step**:
+   ```bash
+   npm run e2etest:debug
+   ```
+
+4. **Run specific test file**:
+   ```bash
+   npx playwright test src/simulations/scenario1.spec.ts
+   ```
+
+5. **Run tests in headed mode (visible browser)**:
+   ```bash
+   npx playwright test --headed
+   ```
+
+6. **Run unit tests**:
+   ```bash
+   npm run unittest:run      # Run once
+   npm run unittest:watch    # Run in watch mode
+   npm run unittest:ui       # Run with UI
+   ```
+
+#### Test Configuration
+
+Tests are configured in `playwright.config.ts`:
+```typescript
+export default defineConfig({
+  testDir: './src/simulations',
+  outputDir: './e2etest-results',
+  fullyParallel: true,
+  use: {
+    trace: 'off',
+  },
+  projects: [
+    {
+      name: 'chromium',
+      use: {...devices['Desktop Chrome']},
+    },
+  ],
+});
+```
+
+#### Creating New Test Scenarios
+
+To create a new test scenario:
+
+1. **Create the implementation file** (`src/simulations/scenarioN.ts`):
+   ```typescript
+   export async function scenarioN({page, userId, password}: BrowserInstance, serverURL: string) {
+     // Implementation for load testing (infinite loop)
+   }
+   ```
+
+2. **Create the test specification file** (`src/simulations/scenarioN.spec.ts`):
+   ```typescript
+   import {test} from '@playwright/test';
+   
+   test('Scenario N', async ({page}) => {
+     // Single-run test for manual validation
+   });
+   ```
+
+3. **Update the browser manager** to include the new scenario in the test execution.
+
+### Test Results and Reports
+
+- Test results are stored in `e2etest-results/`
+- Playwright generates detailed HTML reports
+- Screenshots and videos are captured on failures
+- Trace files can be generated for debugging
+
+## Development
+
+### Available Make Commands
+
+The project uses a Makefile for build and development operations:
+
+```bash
+make help                 # Show all available commands
+make install             # Install dependencies and Playwright browser
+make install-dependencies # Install only Node.js dependencies
+make install-playwright  # Install Playwright Chromium browser
+make build              # Build the TypeScript server
+make start              # Start server in production mode
+make start-dev          # Start server in development mode
+make start-watch        # Start server in watch mode (auto-restart)
+make clean              # Clean build artifacts and dependencies
+```
```

Comment:
I suggest we document the commands in the Makefile itself and here only say that `make help` will list them all.

Reply:
removed